src/drive/storage/index.ts

```diff
@@ -0,0 +1,381 @@
+        throw new Error(extractAxiosErrorMessage(error));
+      });
+
+    return [promise, cancelTokenSource];
+  }
+
+  /**
+   * Moves a specific folder to a new location
+   * @param payload
+   * @param hashPath
+   */
+  public async moveFolder(
+    payload: MoveFolderPayload,
+    hashPath: HashPath
+  ): Promise<MoveFolderResponse> {
+    const moveFolderResponse = await this.axios
+      .post(`${this.apiUrl}/api/storage/move/folder`, {
+        folderId: payload.folder.id,
+        destination: payload.destinationFolderId,
+      }, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+
+    const finalFolderPath = `${payload.destinationPath}/${payload.folder.name}`;
+    await this.updateFolderContents(
+      payload.folder.id,
+      finalFolderPath,
+      payload.bucketId,
+      hashPath
+    );
+
+    return moveFolderResponse;
+  }
+
+  /**
+   * Updates the metadata of a folder
+   * @param payload
+   * @param hashPath
+   */
+  public async updateFolder(
+    payload: UpdateFolderMetadataPayload,
+    hashPath: HashPath
+  ): Promise<void> {
+    await this.axios
+      .post(`${this.apiUrl}/api/storage/folder/${payload.folderId}/meta`, payload.changes, {
+        headers: this.headers()
+      })
+      .then(response => {
+        return response.data;
+      });
+
+    await this.updateFolderContents(
+      payload.folderId,
+      payload.destinationPath,
+      payload.bucketId,
+      hashPath
+    );
+  }
+
+  /**
+   * Fetches & returns the contents of a specific folder
+   * @param folderId
+   */
+  public getFolderContent(folderId: number): [
+    Promise<{
+      folders: DriveFolderData[];
+      files: DriveFileData[]
+    }>,
+    CancelTokenSource
+  ] {
+    const cancelTokenSource = axios.CancelToken.source();
+    const promise = this.axios
+      .get<FetchFolderContentResponse>(`${this.apiUrl}/api/storage/v2/folder/${folderId}`, {
+        cancelToken: cancelTokenSource.token,
+        headers: this.headers()
+      })
+      .then(response => {
+        return {
+          folders: response.data.children.map(folder => ({ ...folder, isFolder: true })),
+          files: response.data.files,
+        };
+      })
+      .catch(error => {
+        throw new Error(extractAxiosErrorMessage(error));
+      });
+
+    return [promise, cancelTokenSource];
+  }
+
+  /**
+   * Removes a specific folder from the centralized persistence
+   * @param folderId
+   */
+  public deleteFolder(folderId: number): Promise<void> {
+    return this.axios
+      .delete(`${this.apiUrl}/api/storage/folder/${folderId}`, {
+        headers: this.headers()
+      });
+  }
+
+  /**
+   * Updates all the elements contained in a specific folder
+   * @param folderId
+   * @param finalPath
+   * @param bucketId
+   * @param hashPath
+   * @private
+   */
+  private async updateFolderContents(folderId: number, finalPath: string, bucketId: string, hashPath: HashPath) {
```

Comment:
This is no longer needed, the last Desktop release uses UUIDs as filenames and Drive-Web does this too. Check [this](https://github.com/internxt/drive-web/pull/251) @sraver